### PR TITLE
Doc Ruby Tests

### LINKED ISSUE #136: Document how to use Ruby interface
See the discussion in https://github.com/ignitionrobotics/ign-math/issues/134#issuecomment-659048663.

We have Ruby bindings, but it's not easy to find out how to use them.

Things that need to be documented:

* Requirements (i.e. `ruby-dev` and `swig`)
* Usage
* How to test

### PR DESCRIPTION
**Pull request by luccosta**
Resolves #136 .

Add install prerequisites and testing subsection Ruby Tests.

Tried to do an guide for absolute beginners, on how to create an interface and use this in a test file, using `Swig` documentation important points as reference.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -90,6 +90,20 @@ The optional Eigen component of Ignition Math requires:
     sudo apt-get install libeigen3-dev
     ```
 
+The optional Ruby tests of Ignition Math requires:
```
Comment (chapulina) [inline README.md:93]:
```suggestion
The optional Ruby tests of Ignition Math require:
```

File: README.md [hunk 2]
```diff
@@ -90,6 +90,20 @@ The optional Eigen component of Ignition Math requires:
...
+
+ * [Ruby](https://www.ruby-lang.org/). Refer to the [Ruby Documentation](https://www.ruby-lang.org/downloads/) for installation instructions. On Ubuntu Systems `apt-get` can be used to install Ubuntu Package `ruby-dev`:
```
Comment (chapulina) [inline README.md:95]:
```suggestion
 * [Ruby](https://www.ruby-lang.org/). Refer to the [Ruby Documentation](https://www.ruby-lang.org/downloads/) for installation instructions. On Ubuntu systems `apt-get` can be used to install Ubuntu Package `ruby-dev`:
```

File: README.md [hunk 3]
```diff
@@ -90,6 +90,20 @@ The optional Eigen component of Ignition Math requires:
...
+
+    ```
+    sudo apt-get install ruby-dev
+    ```
+
+  * [Swig](http://www.swig.org/). Refer to the [Swig Documentation](http://www.swig.org/download.html) for installation instructions. On Ubuntu Systems `apt-get` can be used to install Swig:
```
Comment (chapulina) [inline README.md:101]:
```suggestion
  * [Swig](http://www.swig.org/). Refer to the [Swig Documentation](http://www.swig.org/download.html) for installation instructions. On Ubuntu systems `apt-get` can be used to install Swig:
```

File: README.md [hunk 4]
```diff
@@ -162,6 +176,28 @@ Follow these steps to run tests and static code analysis in your clone of this r
     make codecheck
     ```
 
+## Ruby Tests
+
+### Usage
+
+The C++ classes are available in Ruby code by interfaces files (.i) used by swig to build a C++ extension module.
```
Comment (chapulina) [inline README.md:183]:
```suggestion
The C++ classes are available in Ruby code by interface files (`.i`) used by swig to build a C++ extension module.
```

File: README.md [hunk 5]
```diff
@@ -162,6 +176,28 @@ Follow these steps to run tests and static code analysis in your clone of this r
...
+
+The interfaces and Ruby test codes are in `src` folder. To use a C++ class in Ruby you need:
```
Comment (chapulina) [inline README.md:185]:
```suggestion
The interfaces and Ruby test codes are in the `src` folder. To use a C++ class in Ruby you need to:
```

File: README.md [hunk 6]
```diff
@@ -162,6 +176,28 @@ Follow these steps to run tests and static code analysis in your clone of this r
...
+
+1. Create an interface file describing the class as in Swig and Ruby reference at [The Ruby-to-C/C++ Mapping](http://www.swig.org/Doc1.3/Ruby.html#Ruby_nn11)
+
+2. Include the interface file in `/src/ing_math.i`
+
+3. Create the Ruby file and import the class as in Swig and Ruby reference at [C++ Classes](http://www.swig.org/Doc1.3/Ruby.html#Ruby_nn18)
+
+### Tests
+
+`make test` already run all tests, including the ones made in Ruby, but you can run an Ruby test individually using
```
Comment (chapulina) [inline README.md:195]:
```suggestion
`make test` already runs all tests, including the ones made in Ruby, but you can run a Ruby test individually using
```

**Comment by luccosta** [comment]
Thanks for the suggestions! I'm getting into the use of `this` :sweat_smile: .

**Comment by chapulina** [comment]
> Thanks for the suggestions! I'm getting into the use of this sweat_smile . I accepted all.

No worries. One annoying thing when using the GitHub UI to accept suggestions is that they're not signed by default. You need to add the signature on the UI yourself, which is inconvenient :confused: That's why the DCO check failed... But don't worry this time, I'll make sure the commit is signed when squash-merging.

**Pull request merged**